test/unit/test-amp-story-player.js

```diff
@@ -21,23 +21,38 @@ import {toArray} from '../../src/types';
 describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
   let win;
   let playerEl;
+  let playerSpy;
 
   const fireHandler = [];
-  const DEFAULT_URL =
+  const DEFAULT_CACHE_URL =
     'https://www-washingtonpost-com.cdn.ampproject.org/v/s/www.washingtonpost.com/graphics/2019/lifestyle/travel/amp-stories/a-locals-guide-to-what-to-eat-and-do-in-new-york-city/';
+  const DEFAULT_ORIGIN_URL =
+    'https://www.washingtonpost.com/graphics/2019/lifestyle/travel/amp-stories/a-locals-guide-to-what-to-eat-and-do-in-new-york-city/';
   let fakeMessaging;
   let messagingMock;
   let oldPrototype;
 
-  function buildStoryPlayer(numStories = 1, url = DEFAULT_URL) {
+  async function buildStoryPlayer(
+    numStories = 1,
+    url = DEFAULT_CACHE_URL,
+    cache = null
+  ) {
     playerEl = win.document.createElement('amp-story-player');
+
+    if (cache) {
+      playerEl.setAttribute('amp-cache', cache);
+    }
     for (let i = 0; i < numStories; i++) {
       const storyAnchor = win.document.createElement('a');
       storyAnchor.setAttribute('href', url);
       playerEl.appendChild(storyAnchor);
     }
 
     win.document.body.appendChild(playerEl);
+
+    for (let i = 0; i < numStories; i++) {
+      await Promise.resolve(); // Microtask tick.
+    }
```

Comment:
Can you try to use lolex and the `runMicrotasks()` method?
https://openbase.io/js/lolex

If it works we should switch to that for all the new AMP story tests we write.